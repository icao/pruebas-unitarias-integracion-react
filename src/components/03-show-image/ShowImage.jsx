import React from 'react'
import './ShowImage.scss'
import PropTypes from 'prop-types'

const ShowImage = ({ url, title }) => {
  return (
    <div className='card animate__animated animate__fadeInDown'>
      <h3>Consumiendo API Giphy - Gif Random</h3>
      <img className='card__image' src={url} alt={title} />
      <h4>{title}</h4>
    </div>
  )
}

ShowImage.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}
export default ShowImage
