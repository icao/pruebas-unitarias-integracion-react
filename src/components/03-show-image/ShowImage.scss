@import '../../scss/Palette';

.card {
  width: 450px;
  height: auto;
  border: 1px solid $color-white;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;

  &__image {
    width: 100%;
    height: auto;
  }
}
