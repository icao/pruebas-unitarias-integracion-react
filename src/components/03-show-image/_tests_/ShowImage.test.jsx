import { shallow } from 'enzyme'
import React from 'react'
import ShowImage from '../ShowImage'

describe('Probando el componente <ShowImage />', () => {
  const url = 'https://media.giphy.com/media/d8j1B7L3sLmO6rdFAA/giphy.gif'
  const titulo = 'Super Pibu'
  const wrapper = shallow(<ShowImage url={url} title={titulo} />)

  test('El componente <ShowImage /> debe mostrarse correctamente y aceptar las propiedades url y titulo', () => {
    expect(wrapper).toMatchSnapshot()
  })
  // TODO: El texto(titulo) renderizado debe ser el mismo a la prop titulo
  // TODO: El atributo 'src' debe ser igual a la prop url, y el 'atl' igual a la prop titulo
})
