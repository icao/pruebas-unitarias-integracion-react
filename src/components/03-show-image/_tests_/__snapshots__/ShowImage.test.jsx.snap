// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <ShowImage /> El componente <ShowImage /> debe mostrarse correctamente y aceptar las propiedades url y titulo 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <h3>
    Consumiendo API Giphy - Gif Random
  </h3>
  <img
    alt="Super Pibu"
    className="card__image"
    src="https://media.giphy.com/media/d8j1B7L3sLmO6rdFAA/giphy.gif"
  />
  <h4>
    Super Pibu
  </h4>
</div>
`;
